common/config-parser/src/types/spec.rs

```diff
@@ -292,7 +305,7 @@ impl From<HarforkInput> for HardforkInfoInner {
     }
 }
 
-#[derive(Clone, Debug, Deserialize, Copy)]
-enum HardforkType {
+#[derive(Clone, Debug, Serialize, Deserialize, Copy, ValueEnum, EnumIter, PartialEq, Eq, Hash)]
+pub enum HardforkType {
```

Comment:
Perhaps named `HardforkName` is more suitable.

```suggestion
pub enum HardforkName {
```